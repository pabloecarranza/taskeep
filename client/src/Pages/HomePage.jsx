import React from 'react';
import { Sidebar } from './../components/Sidebar';
import { TaskArea } from './../components/TaskArea';
import { Box, Flex } from '@chakra-ui/react';

export const HomePage = () => {
	return (
		<Flex justifyContent='center' mt='5px'>
			<Sidebar />
			<TaskArea />
		</Flex>
	);
};
