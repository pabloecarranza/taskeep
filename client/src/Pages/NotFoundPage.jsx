import {
	Heading,
	Container,
	Box,
	Center,
	Button,
	Text,
	Image,
} from '@chakra-ui/react';
import Logo from '../assets/daily-tasks.png';
import { ArrowForwardIcon } from '@chakra-ui/icons';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';

export const NotFoundPage = ({
	error_text,
	message_redirect,
	page_redirect,
	test = false,
}) => {
	let navigate = useNavigate();
	const reDirectTo = () => {
		navigate('/homepage', { replace: true });
	};

	return (
		<Container centerContent>
			<Center padding='4' w='100vw' h='100vh'>
				<Box
					w='80%'
					h='80%'
					display='flex'
					flexDir='column'
					alignItems='center'
					justifyContent='center'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Heading as='h2' size='2xl' fontWeight='bold' textAlign='center'>
						{error_text}
						<Text
							bgGradient={test ? '' : 'linear(to-l, #7928ca, #0084ff)'}
							bgClip='text'
							fontSize='3xl'
							fontWeight='bold'
							pb='30px'
							pt='20px'
						>
							{message_redirect}
						</Text>
					</Heading>
					<Box>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							size='lg'
							onClick={reDirectTo}
						>
							{page_redirect}
						</Button>
					</Box>
				</Box>
			</Center>
		</Container>
	);
};

NotFoundPage.propTypes = {
	error_text: PropTypes.string.isRequired,
	message_redirect: PropTypes.string.isRequired,
	page_redirect: PropTypes.string.isRequired,
};
