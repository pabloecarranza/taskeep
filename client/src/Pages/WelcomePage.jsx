import {
	Box,
	Flex,
	Text,
	Heading,
	Button,
	Center,
	Image,
	useDisclosure,
	ScaleFade,
	Link,
} from '@chakra-ui/react';
import { ExternalLinkIcon } from '@chakra-ui/icons';
import { BsLinkedin, VscGithubInverted } from 'react-icons/all';
import Logo from '../assets/daily-tasks.png';
import Typewriter from 'typewriter-effect';
import { LoginForm } from '../components/LoginForm';
import { ArrowForwardIcon } from '@chakra-ui/icons';

import bg from '../assets/hojas.png';

export const WelcomePage = () => {
	const { isOpen, onToggle } = useDisclosure();

	return (
		<Flex
			w='100vw'
			h='100vh'
			flexDir={{ base: 'column', md: 'row', lg: 'row' }}
		>
			<Box w='100%' h='100%'>
				<Center
					w='100%'
					h='100%'
					flexDir='column'
					alignItems='flex-end'
					textAlign='right'
					pr='35px'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Text
						bgGradient='linear(to-l, #b785e9, #61acf3)'
						bgClip='text'
						fontSize='6xl'
						fontWeight='extrabold'
						pb='10px'
					>
						TASKEEP
					</Text>
					<Heading as='h2' size='2xl' noOflines={1} fontWeight='bold'>
						<Typewriter
							options={{
								strings: ['The most powerful', 'The real powerful'],
								autoStart: true,
								loop: true,
							}}
						/>

						<Text
							bgGradient='linear(to-l, #7928ca, #0084ff)'
							bgClip='text'
							fontSize='5xl'
							fontWeight='bold'
							pb='50px'
						>
							Task Management App
						</Text>
					</Heading>
					<Box>
						<Button
							size='lg'
							colorScheme='blue'
							rightIcon={<VscGithubInverted />}
							mr='10px'
							onClick={() =>
								window.open(
									'https://github.com/pabloecarranza/taskeep',
									'_blank'
								)
							}
						>
							GitHub
						</Button>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							variant='outline'
							size='lg'
							onClick={onToggle}
						>
							Get Started
						</Button>
					</Box>

					<Button
						colorScheme='gray'
						variant='link'
						rightIcon={<BsLinkedin />}
						pt='35px'
						onClick={() =>
							window.open(
								'https://www.linkedin.com/in/pabloecarranza/',
								'_blank'
							)
						}
					>
						Made by Pablo Carranza
					</Button>
				</Center>
			</Box>
			<Center
				w='80%'
				h='100%'
				backgroundImage={bg}
				backgroundPosition='center'
				backgroundRepeat='no-repeat'
				backgroundSize='cover'
			>
				<ScaleFade initialScale={0.9} in={isOpen}>
					<LoginForm />
				</ScaleFade>
			</Center>
		</Flex>
	);
};
