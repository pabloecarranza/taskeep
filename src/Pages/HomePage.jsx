import { Flex } from '@chakra-ui/react';
import { useEffect } from 'react';
import { Sidebar } from '../components/Sidebar/Sidebar';
import { MainArea } from './../components/Routes/MainArea';
import { useDispatch } from 'react-redux';
import { sessionIn, sessionOut } from '../features/api/sessionSlice';
import { useNavigate } from 'react-router-dom';
export const HomePage = () => {
	const userData = JSON.parse(localStorage.getItem('identified-user'));
	const dispatch = useDispatch();
	const navigate = useNavigate();
	const userCheck = () => {
		if (!userData) {
			dispatch(sessionOut());
			navigate('/');
			return;
		}
		dispatch(sessionIn(userData));
	};

	useEffect(() => {
		userCheck();
	}, []);

	return (
		<Flex justifyContent='center' mt='5px'>
			<Sidebar />
			<MainArea />
		</Flex>
	);
};
