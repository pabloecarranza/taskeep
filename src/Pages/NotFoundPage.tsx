import {
	Heading,
	Container,
	Box,
	Center,
	Button,
	Text,
	Image,
} from '@chakra-ui/react';
import Logo from '../assets/daily-tasks.png';
import { ArrowForwardIcon } from '@chakra-ui/icons';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';

interface Props {
	errorText: string,
	messageRedirect: string,
	pageRedirect: string,
	test: boolean,
}

export const NotFoundPage = ({
	errorText,
	messageRedirect,
	pageRedirect,
	test = false,
}: Props ) => {
	const navigate = useNavigate();
	const reDirectTo = () => {
		navigate('/', { replace: true });
	};

	return (
		<Container centerContent>
			<Center padding='4' w='100vw' h='100vh'>
				<Box
					w='80%'
					h='80%'
					display='flex'
					flexDir='column'
					alignItems='center'
					justifyContent='center'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Heading as='h2' size='2xl' fontWeight='bold' textAlign='center'>
						{errorText}
						<Text
							bgGradient={test ? '' : 'linear(to-l, #7928ca, #0084ff)'}
							bgClip='text'
							fontSize='3xl'
							fontWeight='bold'
							pb='30px'
							pt='20px'
						>
							{messageRedirect}
						</Text>
					</Heading>
					<Box>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							size='lg'
							onClick={reDirectTo}
						>
							{pageRedirect}
						</Button>
					</Box>
				</Box>
			</Center>
		</Container>
	);
};

NotFoundPage.propTypes = {
	errorText: PropTypes.string.isRequired,
	messageRedirect: PropTypes.string.isRequired,
	pageRedirect: PropTypes.string.isRequired,
	test: PropTypes.bool,
};
