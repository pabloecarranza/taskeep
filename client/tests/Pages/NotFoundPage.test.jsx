import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { expect, describe, it } from 'vitest';
import { NotFoundPage } from './../../src/Pages/NotFoundPage';

describe('Test suite on NotFoundPage component', () => {
	const NotFoundPageDates = {
		error_text: "We're",
		message_redirect: 'Our team',
		page_redirect: 'Page',
	};

	it('must show the error text app send by props', () => {
		const { getByText } = render(
			<Router>
				<NotFoundPage {...NotFoundPageDates} />
			</Router>
		);
		expect(getByText(NotFoundPageDates.error_text)).toBeTruthy();
	});

	it('must show the redirect message send by props', () => {
		const { getByText } = render(
			<Router>
				<NotFoundPage {...NotFoundPageDates} />
			</Router>
		);
		expect(getByText(NotFoundPageDates.message_redirect));
	});

	it('must show the name of page to redirect  send by props', () => {
		const { getByText } = render(
			<Router>
				<NotFoundPage {...NotFoundPageDates} />
			</Router>
		);
		expect(getByText(NotFoundPageDates.page_redirect));
	});

	it('must match vith the snapshot', () => {
		const { container } = render(
			<Router>
				<NotFoundPage {...NotFoundPageDates} />
			</Router>
		);
		expect(container).toMatchSnapshot();
	});
});
