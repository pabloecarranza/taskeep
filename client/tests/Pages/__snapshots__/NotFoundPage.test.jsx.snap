// Vitest Snapshot v1

exports[`Test suite on NotFoundPage component > must match vith the snapshot 1`] = `
<div>
  <div
    class="chakra-container css-dvxtzn"
  >
    <div
      class="css-hmqmvr"
    >
      <div
        class="css-1wf37dc"
      >
        <div
          class="css-0"
        >
          <img
            alt="Logo_Task"
            class="chakra-image css-50xotc"
            src="/src/assets/daily-tasks.png"
          />
        </div>
        <h2
          class="chakra-heading css-1akq1da"
        >
          We're
          <p
            class="chakra-text css-ejz6fe"
          >
            Our team
          </p>
        </h2>
        <div
          class="css-0"
        >
          <button
            class="chakra-button css-4xx2wk"
            type="button"
          >
            Page
            <span
              class="chakra-button__icon css-1hzyiq5"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
